File: src/main/java/ru/spb/sspk/ssdmd/phonebook_test/repository/PersonRepository.java
package ru.spb.sspk.ssdmd.phonebook_test.repository;

import ru.spb.sspk.ssdmd.phonebook_test.model.entity.Person;

import java.util.List;
import java.util.Map;

public interface PersonRepository {


    Person save(Person person);

    Person update(Person person);

    List<Person> findAll();

    List<Person> findByAll(Map<String, Object> paramMap);

    void deleteById(Long id);
}

File: src/main/java/ru/spb/sspk/ssdmd/phonebook_test/repository/PersonRepository.kt
package ru.spb.sspk.ssdmd.phonebook_test.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param
import org.springframework.stereotype.Repository
import ru.spb.sspk.ssdmd.phonebook_test.model.entity.Person

@Repository
interface PersonRepository : JpaRepository<Person, Long> {
    @Query(
        "SELECT p FROM Person p WHERE p.firstname = :firstname " +
                "OR p.lastname = :lastname " +
                "OR p.department = :department " +
                "OR p.block = :block " +
                "OR p.sector = :sector " +
                "OR p.mobilPhone = :mobilPhone " +
                "OR p.phone = :phone"
    )
    fun findByFirstnameOrLastnameOrDepartmentOrBlockOrSectorOrMobilPhoneOrPhone(
        @Param("firstname") firstname: String?,
        @Param("lastname") lastname: String?,
        @Param("department") department: String?,
        @Param("block") block: String?,
        @Param("sector") sector: String?,
        @Param("mobilPhone") mobilPhone: String?,
        @Param("phone") phone: String?
    ): List<Person>
}